### DISCUSSION TITLE
Require up to date reviews for CODEOWNERS but nothing else

### DISCUSSION THREAD

**Discussion by ekmartin** [community/community#48944] [Pull Requests]
### Select Topic Area

Question

### Body

I have a repository where `main` is a protected branch without required code reviews—i.e., as long as a PR is opened the author can merge it themselves. I would like to keep it that way, but now I want to add required code reviews for one specific folder. I thought I could do this with `CODEOWNERS` but I haven't found an incantation that does exactly what I need yet.

The closest I've come is this:
<img width="751" alt="image" src="https://user-images.githubusercontent.com/3536982/222321929-b2ce0fcb-d3ee-4f02-a131-abcd755f90d6.png">

But then you only need an approval for the PR once: if you push a completely unrelated commit, you don't need another review. Naturally that's what the `Require approval of the most recent reviewable push` option is for, but if I enable this then suddenly _other_ branches that aren't affected by `CODEOWNERS` also need reviews:

<img width="674" alt="image" src="https://user-images.githubusercontent.com/3536982/222322130-3982b5ef-51c3-43e7-9dbd-c3906c709fbd.png">

Is there any way to do this with standard GitHub tooling or would I have to write a custom action?

**Comment by Anmol-Baranwal** [comment]
You cannot add it for a specific folder, you can only do it for a branch. 
I don't think writing a github action would help either.
Although if you want the code owner to push without needing a review then there is a option of `bypass` which you can check.

**Reply by Anmol-Baranwal** [reply]
Do correct me, if I am wrong.

**Reply by Anmol-Baranwal** [reply]
So, if there are no further doubts. Could you mark my answer as answered and let others know that question has been answered.

**Reply by ekmartin** [reply]
I'm not sure what you're referring to with branch here, but I found that this works as intended if I use the `Dismiss stale pull request approvals`.

**Comment by ekmartin** [comment]
It looks like this works as intended with `Dismiss stale pull request approvals`.
